crates/miden-lib/asm/account_components/multisig_rpo_falcon_512.masm

```diff
@@ -293,7 +298,87 @@ export.auth_tx_rpo_falcon512_multisig
 
     # ------ Checking threshold is >= num_verified_signatures ------
 
+    # 1. initialize transaction_threshold = 0
+    # 2. iterate through all account procedures
+    #   a. check if the procedure was called during the transaction
+    #   b. if called, check if this procedure was included in the proc_threshold_map
+    #   c. if included, get its proc_threshold
+    #   d. if proc_threshold > transaction_threshold, set transaction_threshold = proc_threshold
+    # 3. if transaction_threshold == 0 at the end, revert to using default_threshold
+
     movup.5
+    # => [default_threshold, num_verified_signatures, TX_SUMMARY_COMMITMENT]
+
+    # store default_threshold for later
+    loc_store.0
+    # => [num_verified_signatures, TX_SUMMARY_COMMITMENT]
+
+    # initialize transaction_threshold = 0
+    push.0
+    # => [transaction_threshold, num_verified_signatures, TX_SUMMARY_COMMITMENT]
+
+    # get the number of account procedures
+    exec.account::get_num_procedures
+    # => [num_procedures, transaction_threshold, num_verified_signatures, TX_SUMMARY_COMMITMENT]
+
+    # for each procedure, see if this procedure has been called in the transaction.
+    dup neq.0
+    # => [should_continue, num_procedures, transaction_threshold, num_verified_signatures, TX_SUMMARY_COMMITMENT]
+    while.true
+        sub.1 dup
+        # => [num_procedures-1, num_procedures-1, transaction_threshold, num_verified_signatures, TX_SUMMARY_COMMITMENT]
+
+        # get procedure root of the procedure with index i
+        exec.account::get_procedure_root dupw
+        # => [PROC_ROOT, PROC_ROOT, num_procedures-1, transaction_threshold, num_verified_signatures, TX_SUMMARY_COMMITMENT]
+
+        # see if this procedure has been called in the transaction.
+        exec.account::was_procedure_called
+        # => [was_called, PROC_ROOT, num_procedures-1, transaction_threshold, num_verified_signatures, TX_SUMMARY_COMMITMENT]
+
+        # if it has been called, get the threshold of that LenientProcedure. Call it current_threshold
+        if.true
+            # => [PROC_ROOT, num_procedures-1, transaction_threshold, num_verified_signatures, TX_SUMMARY_COMMITMENT]
+
+            push.PROC_THRESHOLD_ROOTS_SLOT
+            # => [PROC_THRESHOLD_ROOTS_SLOT, PROC_ROOT, num_procedures-1, transaction_threshold, num_verified_signatures, TX_SUMMARY_COMMITMENT]
+
+            exec.account::get_map_item
+            # => [[0, 0, 0, proc_threshold], num_procedures-1, transaction_threshold, num_verified_signatures, TX_SUMMARY_COMMITMENT]
+
+            drop drop drop dup dup.3
+            # => [transaction_threshold, proc_threshold, proc_threshold, num_procedures-1, transaction_threshold, num_verified_signatures, TX_SUMMARY_COMMITMENT]
+
+            gt
```

Comment:
We should be able to use `u32gt` here, right? It'll make the code a bit more efficient.

Reply:
done in [chore: use u32gt instead of gt for efficiency](https://github.com/0xMiden/miden-base/pull/1968/commits/74ba40737e0285d1d2db846bb1076ec6a98b173e)